### DISCUSSION TITLE
Where is Constants?

### DISCUSSION THREAD

**Discussion by EmrecanKaracayir** [microsoft/CsWin32#480] [Q&A]
CS0234	The type or namespace name 'Constants' does not exist in the namespace 'Windows.Win32' (are you missing an assembly reference?)

**Comment by AArnott** [comment]
As of #444 which shipped in 0.1.619-beta, we no longer generate a `Constants` class. Constants are instead generated into the same `PInvoke` class with the `extern` methods. 
I'll update the readme to reflect this.

**Reply by lostmsu** [reply]
This problem is back. Where do I find `DXGI_ERROR_NOT_FOUND` with Microsoft.Windows.CsWin32 0.3.106? Or worse, `CM_GETIDLIST_FILTER_CLASS`?

In general, it would be nice to have some discoverability. All the functions go into PInvoke, but other kinds of constructs are all over the place, and it is not obvious how to locate them.

**Reply by AArnott** [reply]
There are two common techniques I use:
1. While in NativeMethods.txt, place the caret on the symbol I want to find and press Ctrl+T (i.e. the Go to All command). That'll search for the symbol, usually including the generated files, and help you find where it is declared.
2. Expand the Dependencies tree to find the generated files and look for it there.
![image](https://github.com/user-attachments/assets/5d8b7a15-19b7-423a-bd21-16eb84bacdb0)

**Reply by lostmsu** [reply]
The Ctrl+T approach works, but you need to enable "Search in external items" toggle
![image](https://github.com/user-attachments/assets/7807c980-a071-4691-bbe1-6d7a994f236a)

**Reply by lostmsu** [reply]
Any idea if the same thing is possible in Rider?

**Reply by AArnott** [reply]
No idea. I don't have a Rider license.
